app/models/person.rb

```diff
@@ -82,6 +82,13 @@ def category
     @category ||= Categories::FindByNomisBookingId.new(latest_nomis_booking_id).call
   end
 
+  def update_nomis_data
+    People::RetrieveImage.call(self, force_update: true)
+    return if @prison_number.blank?
+
+    Profiles::ImportAlertsAndPersonalCareNeeds.new(@profile, @prison_number).call
```

Comment:
I think this needs a bit of rejigging:
1. you probably want to refresh `person.latest_nomis_booking_id` _before_ retrieving the person's image (however see point 3 below)
2. a `Person` can have multiple `Profiles`. The variable `@profile` here isn't defined and so will always be nil
3. more generally, there is a complicated dance the frontend does with the backend: searching for a person by prison_number will _create_ new person records from Nomis data. And creating a blank profile for a person with a prison_number will fetch the latest Nomis alerts/data, prior to creating the move. So possibly this step is redundant because it would have have already been called for someone creating the move through BaSM front end. See this walkthrough of the typical interactions between FE and BE: https://github.com/ministryofjustice/hmpps-book-secure-move-api/wiki/API-Walkthroughs-7.1-Prison-to-Court-move

Reply:
in short, it might not be necessary to call `Profiles::ImportAlertsAndPersonalCareNeeds` - the only refresh required might be of the person's image. But this depends on the use-case: whether the intention is to improve data quality on moves created via the front end, or to improve data quality on moves created elsewhere (e.g. directly by suppliers)